packages/ckeditor5-dev-release-tools/lib/tasks/preparerepository.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+const fs = require( 'fs-extra' );
+const upath = require( 'upath' );
+
+/**
+ * @param {Object} options
+ * @param {String} options.outputDirectory Relative path to the destination directory where packages will be stored.
+ * @param {String} [options.cwd] Root of the repository to prepare. `process.cwd()` by default.
+ * @param {String} [options.packagesDirectory] Relative path to a location of packages.
+ * If specified, all of the found packages will be copied.
+ * @param {Array.<String>} [options.packagesToCopy] List of packages that should be processed.
+ * If not specified, all packages found in `packagesDirectory` are considered.
+ * @param {Object} [options.rootPackageJson] Object containing values to use in the created the `package.json` file.
+ * If not specified, the root package will not be created.
+ * @param {String} options.rootPackageJson.name Name of the package. Required value.
+ * @param {Array.<String>} options.rootPackageJson.files Array containing a list of files or directories to copy. Required value.
+ */
+module.exports = async function prepareRepository( options ) {
+	const {
+		outputDirectory,
+		cwd = process.cwd(),
+		packagesDirectory,
+		packagesToCopy,
+		rootPackageJson
+	} = options;
+
+	if ( !rootPackageJson && !packagesDirectory ) {
+		return;
+	}
+
+	const outputDirPath = upath.join( cwd, outputDirectory );
+	await fs.ensureDir( outputDirPath );
+	const outputDirContent = await fs.readdir( outputDirPath );
+
+	if ( outputDirContent.length ) {
+		throw new Error( `Output directory is not empty: "${ outputDirPath }"` );
+	}
+
+	const copyPromises = [];
+
+	if ( rootPackageJson ) {
+		const copyRootItemsPromises = await processRootPackage( {
+			cwd,
+			rootPackageJson,
+			outputDirPath
+		} );
+
+		copyPromises.push( ...copyRootItemsPromises );
+	}
+
+	if ( packagesDirectory ) {
+		const copyPackagesPromises = await processMonorepoPackages( {
+			cwd,
+			packagesDirectory,
+			packagesToCopy,
+			outputDirPath
+		} );
+
+		copyPromises.push( ...copyPackagesPromises );
+	}
+
+	return Promise.all( copyPromises );
+};
+
+async function processRootPackage( { cwd, rootPackageJson, outputDirPath } ) {
```

Comment:
```suggestion
async function processRootPackage( { cwd, packageJson, outputDirectoryPath } ) {
```

I am a big fan of using full names, especially with keys.

Reply:
Also, please add JSDoc to describe input values. Yeah, I miss TS here.